### ISSUE TITLE
Cheatgrass space is set to 0 under future conditions due to issue in existing cheatgrass relative abundance equation 

### ISSUE THREAD

**Issue by kpalmqui** [DrylandEcology/rSFSTEP2#240]

**Comment by kpalmqui** [comment]
Currently we determine cheatgrass relative abundance based on climate of a given site-scenario using the following that was derived from the raw data provided in the SI of Brummer et al. 2016:

```
if prec7 > 30 mm | tmin2 < -13 C,
Annuals_fraction = 0.0
else,
Annuals_fraction = 0.6732229 - 0.0254591prec7 + 0.0538173tmin2 -0.0021601prec7tmin2

if Annuals_fraction < 0,
Annuals_fraction = 0
```

This equation is not working as expected for some sites and results in negative cheatgrass relative abundance.

For sites that have a small positive tmin2 and prec7 below the threshold of 30, cheatgrass relative abundance was consistently negative (and then reset to 0) because of the interaction term. Without the interaction term, these relative abundances are positive.

Example:
prec7 = 26.5, tmin = 1, FRACTION = -0.004
prec7 = 26.5, tmin = 5, FRACTION = -0.02

In addition, several sites with prec7 below the threshold of 30, but slightly negative tmin2 (still fairly warm sites) resulted in negative cheatgrass relative abundance based on the previous equation, which the interaction term could not overcome:

Example:
prec7 =  28, tmin = -1, FRACTION = -0.03

Collectively these issues lead to cheatgrass relative abundance being set to > 0 under current conditions, but negative and ultimately 0 under future conditions, suggesting that cheatgrass would be lost in those sites in the future. This was despite the fact that climate was becoming more suitable for cheatgrass under future conditions due to warming and increases in cool-season precip.

**SOLUTION**

I revisited the Brummer et al. 2016 data and fit an additive linear model to the data for all sites that had prec7 < 35 mm and tmin2 > -13.5. This new equation contains two major changes:

1. Re-evaluation of the thresholds under which cheatgrass relative abundance is set to 0. I made these changes based on careful re-examination of the Figure below. In addition, both of these conditions now need to be true for cheatgrass to be set to 0. 

[BRTErelcover_JulyPPT&MinFebTemp.pdf](https://github.com/DrylandEcology/rSFSTEP2/files/4685115/BRTErelcover_JulyPPT.MinFebTemp.pdf)

2. Removal of the interaction term.

Here is the new implementation:

```
if prec7 > 35 mm && tmin2 < -13.5 C,
Annuals_fraction = 0.0
else,
Annuals_fraction = 0.474506429 - 0.009499188prec7 + 0.023952063tmin2 

if Annuals_fraction < 0,
Annuals_fraction = 0
```

**Comment by kpalmqui** [comment]
This new implementation yields much more reasonable results. 

**Previous implementation:**
- There were 20 sites where the median GCM suggested loss of cheatgrass from current to future conditions for at least one RCP-time period combination.

- An additional 26 sites had cheatgrass being lost in the future under at least 1 GCM, but few enough that the loss of cheatgrass was not the median prediction.

- 6 sites gained cheatgrass from current to future conditions according to the median GCM. These sites were all at relatively high elevation (1743 to 2675 m) cool, wet sites in the Wyoming Basins, Snake River Plain, and Colorado Plateau.

- An additional 10 sites had at least 1 GCM suggesting gain of cheatgrass, but few enough that gain of cheatgrass was not the median prediction. These sites were primarily in the Great Plains.

**New Implementation:**

- 0 sites have lost cheatgrass from current to future conditions

- 31 sites gained cheatgrass from current to future conditions (for at least one RCP-time period combination)

- 48 sites have cheatgrass present under both current and future conditions (all scenarios)

- 12 sites have cheatgrass absent under both current and future conditions (all scenarios)

Here is a Figures that summarizes the site outcomes geographically from current to future conditions:

[Cheatgrass.PA.maps.pdf](https://github.com/DrylandEcology/rSFSTEP2/files/4685134/Cheatgrass.PA.maps.pdf)

**Comment by kpalmqui** [comment]
The new implementation I document above has been revised.

We  (@dschlaep and I) have decided to move forward with a logistic equation that derives cheatgrass relative abundance based on the same two variables above: prec7 (precipitation in July, mm) and tmin2 (minimum temperature in Feb, degrees C). 

I returned to the raw data provided in the SI of Brummer et al. 2016 and fit a logistic (binomial) equation instead. 

We then compared the fit of the linear model with a threshold to the logistic model by generating predictive values from each and plotting those (black, open points) on the raw data (blue points):

[linear.logistic.comparison.pdf](https://github.com/DrylandEcology/rSFSTEP2/files/4708058/linear.logistic.comparison.pdf)

Several issues became apparent as we assessed these alterative approaches.

1. Linear model

- does not represent the maximum cheatgrass relative abundance values in the data well

- still results in negative relative abundance values

2. Logistic model

- results in all values > 0, even under climatic conditions when we expect cheatgrass to be absent (tmin2 < -13.5 & prec7 > 35mm) 

We then remade the above embedded Figure to 1) set negative relative abundance values to 0 for linear approach and 2) set cheatgrass relative abundance to 0 if tmin2 < -13.5 & prec7 > 35mm for the logistic approach. Resulting Figure:

[linear.logistic.comparison.setvaluesto0.pdf](https://github.com/DrylandEcology/rSFSTEP2/files/4708067/linear.logistic.comparison.setvaluesto0.pdf)

We decided to move forward with the logistic approach because it is a better fit to the raw data. We also decided to continue to use the threshold tmin2 < -13.5 & prec7 > 35mm and set cheatgrass relative abundance to 0 if those conditions were met.

It was important to set relative abundance to 0 in sites where Brummer et al. 2016 found no cheatgrass (rather than use the very small positive values derived from the logistic equation), because eind really drives annual biomass (not space). Small positive space values (i.e. 0.0002) and eind = 10 result in fairly high cheatgrass biomass (38 g/m2).

**Comment by kpalmqui** [comment]
Model summary from logistic model:

Call:
glm(formula = data$BRTErelcover ~ data$prec7 + data$tmin2, family = binomial)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.1675  -0.4010  -0.1415   0.2133   1.3286  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)    
(Intercept)  0.80474    0.34614   2.325 0.020075 *  
data$prec7  -0.10017    0.03033  -3.302 0.000959 ***
data$tmin2   0.18181    0.06283   2.894 0.003806 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 185.72  on 554  degrees of freedom
Residual deviance: 110.25  on 552  degrees of freedom
AIC: 229.61

Number of Fisher Scoring iterations: 6

Thus the derived equation to calculate cheatgrass relative abundance is:

1 / (1 + e^-(0.8047441-0.100166*prec7+0.1818125*tmin2))

**Comment by kpalmqui** [comment]
A brief update on the results of the new cheatgrass implementation that @dschlaep  and I settled on – it uses a logistic equation to derive cheatgrass relative abundance based on July precipitation and minimum temperature in February.
 
Cheatgrass relative abundance was > 0 (i.e. cheatgrass on) in all but 5 of the 91 sites under current conditions (grey sites in panel A). Three of those sites are in the Wyoming Basins and the remaining two are high elevation sites (2631 m, 2814 m) in the Colorado Plateau (see Figure below).

[cheatgrass.PA.map.logisticapproach.pdf](https://github.com/DrylandEcology/rSFSTEP2/files/4719151/cheatgrass.PA.map.logisticapproach.pdf)

For the median GCM, all 5 of these sites are more climatically suitable for cheatgrass under all future scenarios (i.e. cheatgrass gained in those sites under future conditions).
 
For one additional site (red site in panel B) in the Wyoming Basins, cheatgrass was represented under current conditions, but only 6 of 13 GCMs had cheatgrass relative abundance > 0 for RCP4.5, mid-century conditions. Thus, the median expectation for cheatgrass relative abundance in this site under that scenario is 0. The relative abundance that were derived under current conditions for this site was very small (< 0.01). 
 
The overall outcome of the new approach is that we are representing cheatgrass in more sites under both current and future conditions than the previous implementation (i.e. linear model).

**Comment by kpalmqui** [comment]
Resolved by bddb281
